package.json

```diff
@@ -120,27 +120,43 @@
         "devskim.ignoreFilesList": {
           "type": "array",
           "default": [
-            "out/*",
-            "bin/*",
-            "node_modules/*",
-            ".vscode/*",
-            "*.lock",
-            "*-lock",
-            "logs/*",
+            "*/out/*",
+            "*/bin/*",
+            "*/.git/*",
+            "*/git/*",
+            "*/node_modules/*",
+            "*/.vscode/*",
+            "*/logs/*",
+            "*/NuGet/*",
+            "*/.cache/*",
+            "*/tests/*",
```

Comment:
Suggest either making these case insensitive, or even better, regular expressions.

Reply:
regular expressions would actually make my life easier, since behind the scenes I convert them to RegExes for processing (initially did globs, but the glob libraries were so bloody slow and I didn't want to recreate them, so just did the wild cards you see).  Do you think requiring a person to be familiar with regular expressions to add to the list is the right usability choice?  Clearly for those of us working on DevSkim regex familiarity is a must, and it would provide a lot of power to people, but for novices (and we do hope novice devs are trying to find security flaws) do you think its ok to ask them to use regular expressions?

case insensitive would be super easy on my part (add an i flag) - do you think there are any scenarios in the *nix world where people would be annoyed by that?

Reply:
You could go with a schema like `/.*\.cache.*/i` meaning "anything that contains .cache, case insensitive", so it's all in one. I think most developers would either be familiar or could learn what they mean pretty easily.

Reply:
Works for me, and in the documentation for the settings I can link to scriptular.com/ to help newbies (and me) make/test their regex.  I'll try to make the change today or Monday (we have an adventure tomorrow).  Also, you and I suck at work/life balance

Reply:
Hey, first pass, what do you think of these Regexes to cover the ignore scenarios:

/\.(exe|dll)$/i
/\.(png|jpg|jpeg|gif|psd|ico|mp3|mpeg)$/i
/\.(zip|tar|gz|rar)$/i
/(\/|\\)(out|bin)(\/|\\)/i
/(\/|\\)(test|test|\_\_tests__|\_\_mocks__)(\/|\\)/i
/(\/|\\)(\.git|git)(\/|\\)/i
/\.(git|git[^\.\\\/]*)$/i
/-lock[^\.\\\/]*|\.lock$/i
/(\/|\\)(\.vscode|\.cache|logs)(\/|\\)/i
/(\/|\\)(nuget|node_modules)(\/|\\)/i
/\.(log|sarif|test)$/i

Reply:
Couple changes, additions

```
/\\.(exe|dll|so|dylib|bin|so\\..*)$/i
/\\.(png|jpg|jpeg|gif|psd|ico|mp3|mpeg|bmp)$/i
/\\.(zip|tar|gz|rar|jar|gz|7z|bz|bz2|gzip|cab|war|xz|nupkg|gem|egg)$/i
/\\.(sqlite3)$/i
/[/\\](out|bin)[/\\]/i
/[/\\](tests?|unittests?|__tests__|__mocks__)[/\\]/i
/[/\\](\\.git|git)[/\\]/i
/\\.(git|git[^.\/])$/i   <-- 
/-lock[^\\\./]|\\.lock$/i
/[/\\](\\.vscode|\\.cache|logs)[/\\]/i
/[/\\](nuget|node_modules)[/\\]/i
/\\.(log|sarif|test)$/i
/\\.(py[cod]$/i
/[/\\]__pycache__[/\\]/i
```

Reply:
the bracket format is way better way of handling slashes.  should \_\_pycache__ be lumped in with node_modules/nuget